code/game/objects/items/weaponry.dm

```diff
@@ -281,35 +281,41 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 	flags_1 = CONDUCT_1
 	force = 9
 	throwforce = 10
-	w_class = WEIGHT_CLASS_NORMAL
+	w_class = WEIGHT_CLASS_BULKY
 	custom_materials = list(/datum/material/iron=1150, /datum/material/glass=75)
 	attack_verb_continuous = list("hits", "bludgeons", "whacks", "bonks")
 	attack_verb_simple = list("hit", "bludgeon", "whack", "bonk")
 
 /obj/item/wirerod/attackby(obj/item/I, mob/user, params)
 	if(istype(I, /obj/item/shard))
-		var/obj/item/spear/S = new /obj/item/spear
-
-		remove_item_from_storage(user)
-		if (!user.transferItemToLoc(I, S))
-			return
-		S.CheckParts(list(I))
-		qdel(src)
+		user.balloon_alert(user, "crafting spear")
+		if(do_after(user, 4 SECONDS, src))
+			var/obj/item/spear/S = new /obj/item/spear
+
+			remove_item_from_storage(user)
+			if (!user.transferItemToLoc(I, S))
+				return
+			S.CheckParts(list(I))
```

Comment:
```suggestion
			var/obj/item/spear/crafted_spear = new /obj/item/spear()

			remove_item_from_storage(user)
			if (!user.transferItemToLoc(I, crafted_spear))
				return
			crafted_spear.CheckParts(list(I))
```
If you clean the `obj/item/I` argument to use the same name as the proc definition you can get a code improvement label.